base/src/com/google/idea/blaze/base/sync/projectstructure/ContentEntryEditor.java

```diff
@@ -100,12 +100,19 @@ private static void walkFileSystem(
     File file = workspaceRoot.fileForPath(workspacePath);
     boolean isTest = testConfig.isTestSource(workspacePath.relativePath());
     SourceFolder current = sourceFolders.get(new File(file.getPath()));
-    SourceFolder currentOrParent = current != null ? current : parent;
-    if (currentOrParent != null && isTest != currentOrParent.isTestSource()) {
-      currentOrParent =
-          provider.setSourceFolderForLocation(contentEntry, currentOrParent, file, isTest);
-      if (current != null) {
+    SourceFolder next;
+    if (current != null) {
```

Comment:
This code contains a lot of branches that look like `if (A != B) { X } else { Y }`. Since the else clause is always present, the code might become more readable if you rewrite it like `if (A == B) { Y } else { X }`.

Reply:
Not entirely sure it's more readable to me, I changed it according to the suggestion though